### PR TITLE
Recreate the content DOM node on location changes

### PR DESCRIPTION
**Pull request by kumar303**
Fixes https://github.com/mozilla/addons-code-manager/issues/1247

### CODE REVIEW

File: src/components/FullscreenGrid/ContentShell.spec.tsx [hunk 1]
```diff
@@ -182,4 +196,11 @@ describe(__filename, () => {
       fullscreenGridActions.toggleAltSidePanel(),
     );
   });
+
+  it('renders the main content panel with a location key', () => {
+    const key = 'locationKey';
+    const root = render({ location: createFakeLocation({ key }) });
+
+    expect(root.find(`.${styles.content}`).key()).toEqual(key);
```
Comment (bobsilverberg) [inline src/components/FullscreenGrid/ContentShell.spec.tsx]: I didn't even know `key()` was a thing! A guess it's rare to need to assert on a `key` in a test.

**Comment by kumar303** [comment]
> Looks good. I'm not sure how `location.key` is calculated (a quick browse of the source didn't make anything obvious), but I assume you're confident that it will be different for each location navigated to.

`location.key` is not [documented](https://reacttraining.com/react-router/web/api/location) well but as far as I can tell this key represents the current location, including the query string and hash. This will change when navigating to different files.

File: src/components/FullscreenGrid/ContentShell.spec.tsx (hunk 1 above)
Comment (kumar303) [reply src/components/FullscreenGrid/ContentShell.spec.tsx]: Heh, yeah, I learned about it recently too. This kind of test relies on trusting the React API but key behavior seems stable enough. It's features like that where having some functional test coverage would be nice.

**Comment by kumar303** [comment]
Thanks for the quick review

**Pull request merged**